src/main/java/uk/gov/hmcts/probate/service/ocr/OCRToCCDMandatoryField.java

```diff
@@ -1,47 +1,118 @@
 import uk.gov.hmcts.probate.model.ccd.ocr.GORMandatoryFields;
+import uk.gov.hmcts.probate.model.ccd.ocr.CaveatMandatoryFields;
 import uk.gov.hmcts.probate.model.ocr.OCRField;
 
-import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.ArrayList;
 import java.util.stream.Stream;
 
 @Slf4j
 @Service
 public class OCRToCCDMandatoryField {
 
+    private static final String MANDATORY_FIELD_WARNING_STIRNG = "Key '%s' is mandatory.";
+    private static final String DEPENDANT_KEY_PRIMARYAPPLICANTALIAS = "primaryApplicantAlias";
+    private static final String DEPENDANT_KEY_IHTREFERENCENUMBER = "ihtReferenceNumber";
+    private static final String DEPENDANT_KEY_IHTFORMID = "ihtFormId";
+    private static final String MANDATORY_KEY_EXECUTORSNOTAPPLYING_EXECUTORNAME =
+            "executorsNotApplying_%s_notApplyingExecutorName";
+    private static final String DEPENDANT_KEY_EXECUTORSNOTAPPLYING_EXECUTORREASON =
+            "executorsNotApplying_%s_notApplyingExecutorReason";
+    private static final String MANDATORY_KEY_PRIMARYAPPLICANTHASALIAS = GORMandatoryFields.PRIMARY_APPLICANT_HAS_ALIAS.getKey();
+    private static final String MANDATORY_KEY_IHTFORMCOMPLETEDONLINE = GORMandatoryFields.IHT_FORM_COMPLETED_ONLINE.getKey();
+
     public List<String> ocrToCCDMandatoryFields(List<OCRField> ocrFields, FormType formType) {
-        List<String> descriptions = new ArrayList<>();
-        List<String> ocrFieldNames = new ArrayList<>();
+        List<String> descriptions = new ArrayList<String>();
+        HashMap<String, String> ocrFieldValues = new HashMap<String, String>();
+
         ocrFields.forEach(ocrField -> {
-            ocrFieldNames.add(ocrField.getName());
+            ocrFieldValues.put(ocrField.getName(), ocrField.getValue());
         });
+
         switch (formType) {
             case PA8A:
                 Stream.of(CaveatMandatoryFields.values()).forEach(field -> {
                     log.info("Checking {} against ocr fields", field.getKey());
-                    if (!ocrFieldNames.contains(field.getKey())) {
+                    if (!ocrFieldValues.containsKey(field.getKey())) {
                         log.warn("{} was not found in ocr fields", field.getKey());
-                        descriptions.add(field.getValue());
+                        descriptions.add(String.format(MANDATORY_FIELD_WARNING_STIRNG, field.getKey()));
                     }
                 });
                 break;
             case PA1A:
+                Stream.of(IntestacyMandatoryFields.values()).forEach(field -> {
+                    log.info("Checking {} against ocr fields", field.getKey());
+                    if (!ocrFieldValues.containsKey(field.getKey())) {
+                        log.warn("{} was not found in ocr fields", field.getKey());
+                        descriptions.add(String.format(MANDATORY_FIELD_WARNING_STIRNG, field.getKey()));
```

Comment:
no longer needs to be description if we are not having descriptions anymore

Reply:
The descriptions List used here is to build the list of warnings and is necessary. The description property on the OCRField isn't necessary, but could be kept so that it can be used once the description field is supported by bulk scanning. Let me know if you would like me to remove this property (from OCRField) and I can update the code.